engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroBaseTransformer.scala

```diff
@@ -87,6 +90,29 @@ trait KafkaAvroBaseTransformer[T] extends SingleInputGenericNodeTransformation[T
     new BasedOnVersionAvroSchemaDeterminer(schemaRegistryClient, preparedTopic.prepared, version)
   }
 
+  protected def topicParamStep(implicit nodeId: NodeId): NodeTransformationDefinition = {
+    case TransformationStep(Nil, _) =>
+      val topicParam = getTopicParam.map(List(_))
+      NextParameters(parameters = topicParam.value, errors = topicParam.written)
+  }
+
+  protected def schemaParamStep(implicit nodeId: NodeId): NodeTransformationDefinition = {
+    case TransformationStep((TopicParamName, DefinedEagerParameter(topic: String, _)) :: Nil, _) =>
+      val preparedTopic = prepareTopic(topic)
+      val versionParam = getVersionParam(preparedTopic)
+      NextParameters(versionParam.value :: paramsDeterminedAfterSchema, errors = versionParam.written)
+    case TransformationStep((TopicParamName, _) :: Nil, _) =>
+      NextParameters(parameters = fallbackVersionOptionParam :: paramsDeterminedAfterSchema)
+  }
+
+  override def initialParameters: List[Parameter] = {
```

Comment:
`(implicit nodeId: NodeId)`?

Reply:
unfortunately no, this method is invoked when there is no node yet :)